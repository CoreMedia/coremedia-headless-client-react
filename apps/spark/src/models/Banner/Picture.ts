import { CmPicture } from "@coremedia-labs/graphql-layer";
import { PreviewMetadata, getPropertyName } from "../../utils/Preview/MetaData";
import { addProperty, mapProperties } from "../../utils/ViewDispatcher/ModelHelper";

export interface SupportsPicture extends PreviewMetadata {
  picture?: Picture;
}

export const addPicture = (self: any, result: SupportsPicture): void => {
  if (self.picture) {
    self.picture && addProperty(result, "picture", initializePicture(self.picture), getPropertyName(self, "picture"));
  }
};

/**
 * @category ViewModels
 */
export interface Picture extends PreviewMetadata {
  title?: string | null;
  alt?: string | null;
  uriTemplate: string | null;
  data: { uri: string | null } | null;
}

/**
 * @category ViewModels
 */
export interface ImageProps {
  picture: Picture;
  cropName?: string;
  width?: number;
  aspectRatio?: string;
}

/**
 * @category ViewModels
 */
export interface ResponsiveImageProps {
  picture: Picture;
  sources: Array<Source>;
}

/**
 * @category ViewModels
 */
export interface Source {
  media?: string;
  srcset: string;
  sizes?: string;
  type?: string;
}

/**
 * Returns a [[Picture]] object based on the GraphQL picture
 * @param picture
 */
export const initializePicture = (picture: CmPicture): Picture => {
  return {
    ...mapProperties(picture, {
      title: "title",
      alt: "alt",
      uriTemplate: "uriTemplate",
      data: "data",
    }),
  };
};
