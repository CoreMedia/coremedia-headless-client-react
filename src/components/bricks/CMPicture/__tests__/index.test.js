// @flow
import React from 'react';

import Picture from '..';

describe('Picture Component', () => {
  it('should render Picture with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Picture
        color="red"
        link="coremedia:///image/2662/data"
        ratio="portrait_ratio1x1"
        title="Scrum"
        theme={{
          breakpoints: {
            small: 480,
            tablet: 768,
            desktop: 992,
            large: 1280,
          },
        }}
      />
    );
    expect(wrapper.shallow()).toMatchSnapshot();
  });
});
