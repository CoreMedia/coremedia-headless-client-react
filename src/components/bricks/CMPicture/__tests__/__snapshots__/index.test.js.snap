// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Component should render Picture with Link, Picture and CTA 1`] = `
<picture>
  <source
    key="0"
    media="(min-width: 1280px)"
    srcSet="http://127.0.0.1:8080/caas/v1/coremedia/sites/caassiopeia-en-DE/media/2662/data?ratio=portrait_ratio1x1&minWidth=1280"
  />
  <source
    key="1"
    media="(min-width: 992px)"
    srcSet="http://127.0.0.1:8080/caas/v1/coremedia/sites/caassiopeia-en-DE/media/2662/data?ratio=portrait_ratio1x1&minWidth=992"
  />
  <source
    key="2"
    media="(min-width: 768px)"
    srcSet="http://127.0.0.1:8080/caas/v1/coremedia/sites/caassiopeia-en-DE/media/2662/data?ratio=portrait_ratio1x1&minWidth=768"
  />
  <source
    key="3"
    media="(min-width: 480px)"
    srcSet="http://127.0.0.1:8080/caas/v1/coremedia/sites/caassiopeia-en-DE/media/2662/data?ratio=portrait_ratio1x1&minWidth=480"
  />
  <Img
    color="red"
    src="http://127.0.0.1:8080/caas/v1/coremedia/sites/caassiopeia-en-DE/media/2662/data?ratio=portrait_ratio1x1&minWidth=320"
    theme={
      Object {
        "colors": Object {
          "default": Object {
            "bgsolid": "rgba(239, 223, 15, 1)",
            "bgtrans": "rgba(239, 223, 15, 0.7)",
            "fg": "#fafafa",
            "huerotation": "5deg",
          },
          "red": Object {
            "bgsolid": "rgba(221, 52, 40, 1)",
            "bgtrans": "rgba(221, 52, 40, 0.7)",
            "fg": "#fafafa",
            "huerotation": "320deg",
          },
        },
      }
    }
    title="Scrum"
  />
</picture>
`;
